base-notebook/start.sh

```diff
@@ -168,12 +168,32 @@ if [ "$(id -u)" == 0 ] ; then
         PATH="${PATH}" \
         PYTHONPATH="${PYTHONPATH:-}" \
         "${cmd[@]}"
-        # Note on the purpose of "PATH=${PATH}":
-        #   In case "${cmd[@]}" is "bash", then PATH will be used by this bash shell.
-        #   However, PATH is irrelevant to how the above sudo command resolves the
-        #   path of "${cmd[@]}". Sudo's path resolution is done via the "secure_path"
-        #   variable set above in /etc/sudoers.d/path.
-
+        # Notes on how we ensure that the environment that this container is started
+        # with is preserved (except vars listed in JUPYTER_ENV_VARS_TO_UNSET) when
+        # we transition from running as root to running as NB_USER.
+        #
+        # - We use `sudo` to execute the command as NB_USER. What then
+        #   happens to the environment will be determined by configuration in
+        #   /etc/sudoers and /etc/sudoers.d/* as well as flags we pass to the sudo
+        #   command. The behavior can be inspected with `sudo -V` run as root.
+        #
+        #   ref: `man sudo`    https://linux.die.net/man/8/sudo
+        #   ref: `man sudoers` https://www.sudo.ws/man/1.8.15/sudoers.man.html
+        #
+        # - We use the `--preserve-env` flag to pass through most environment
+        #   variables, but understand that exceptions are caused by the sudoers
+        #   configuration: `env_delete` and `env_check`.
+        #
+        # - We use the `--set-home` flag to set the HOME variable appropriately.
+        #
+        # - To reduce the default list of variables deleted by sudo, we could have
+        #   used `env_delete` from /etc/sudoers. It has higher priority than the
+        #   `--preserve-env` flag and the `env_keep` configuration.
+        #
+        # - We preserve PATH and PYTHONPATH explicitly. Note however that PATH is
+        #   irrelevant to how the above sudo command resolves the path of `${cmd[@]}`.
+        #   Sudo uses the "secure_path" variable we modified above in
+        #   /etc/sudoers.d/path. (PATH is useful e.g. when `${cmd[@]}` is `bash`.)
```

Comment:
I don't understand the last parenthesis very well, could we rephrase/remove it?

Reply:
Hmm... could you help me to rephrase it?

What I'm trying to say is that if you run `base-notebook start.sh some-command`, then `PATH` is likely completely irrelevant. However, if you run `base-notebook start.sh bash`, then you get a Bash shell with the expected `PATH` set.

Reply:
Hmm... because you assume PATH is irrelevant for a command you start? Maybe... I consider that a quite specific assumption. If you start a jupyter server which is very common, wouldn't the PATH be retained and relevant for the bash terminal sessions run within there for example?

Reply:
Yes, good point. Perhaps it's better to say something like:

> The `PATH` will mainly be relevant for any terminals spawned by the `${cmd[@]}`.

Reply:
I would say something like:

> The PATH will be relevant for any processess spawned by ${cmd[@]} (including new terminals).